scipy/optimize/zeros.py

```diff
@@ -163,54 +163,50 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
         # Newton-Rapheson method
         # Multiply by 1.0 to convert to floating point.  We don't use float(x0)
         # so it still works if x0 is complex.
-        p0 = 1.0 * x0
-        fder2 = 0
+        p0 = 1.0 * asarray(x0)  # convert to ndarray
         for iter in range(maxiter):
             myargs = (p0,) + args
-            fder = fprime(*myargs)
-            if fder == 0:
+            fder = asarray(fprime(*myargs))  # convert to ndarray
+            if (fder == 0).any():
```

Comment:
wouldn't you need to keep iterating the other elements?

Reply:
You could, but IMO you really don't _need_ to. I think it requires indexing which IMO makes the code a little messy. Do you feel very strongly that it should keep iterating the other cases? I'm open to changes.

Reply:
you'd exit early upon hitting a problematic termination condition for any element, even if remaining elements are far from convergence. but I'd defer to a scipy maintainer's opinion here, since non-scalar usage would be new functionality (right? does the released code error on arrays?)

Reply:
Yes. See the traceback in #8354.